CONTRIBUTING.md

```diff
@@ -1,8 +1,123 @@
-# For developers
+# Contributing to PyPose
+Thank you for your interest in contributing to PyPose! 
+Our focus is on creating a state-of-the-art library for robotics research, and we are open to accepting various types of contributions. 
+Before you start coding for PyPose, it is important to inform the PyPose team of your intention to contribute and specify the type of contribution you plan to make. 
+These contributions may include but are not limited to:
 
-## 1. Contributing to Documentation
+1. If you notice a typographical error in PyPose's code or documentation, you can submit a Pull Request directly to fix it.
+    - If the changes you wish to make to PyPose involve significant modifications, it is recommended that you create an issue first. Within the issue, describe the error and information on how to reproduce the bug. Other developers will then review the issue and engage in discussion.
+    - If you require additional information regarding a specific issue, please do not hesitate to request it, and we will gladly provide it to you.
+2. If you have an idea for a new feature in PyPose and wish to implement it, we recommend that you create an issue to post about your proposed feature. From there, we can discuss the design and implementation. Once we agree that the plan looks good, go ahead and implement it.
 
-## 1.1 Build docs locally
+Once you implement and test your feature or bug-fix, please submit a Pull Request. 
+
+## Workflow
+This document covers a step-by-step guide of contributing. It will tell you how to start from setting up the code repository and create a Pull Request step by step. This guide is modified from [OpenMMlab](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md)'s contribution guide.
+
+### 1. Fork and clone
+
+If you are posting a pull request for the first time, you should fork the PyPose repository by clicking the **Fork** button in the top right corner of the GitHub page, and then the forked repositories will appear under your GitHub profile.
+
+Then, you can clone the repositories to local:
+
+```shell
+git@github.com:{your_github_id}/pypose.git
+```
+
+After that, you should add official repository as the upstream repository
+
+```bash
+git remote add upstream git@github.com:pypose/pypose.git
```

Comment:
this doesn't work for me either. how about `git remote add upstream https://github.com/pypose/pypose.git`